lib/matplotlib/contour.py

```diff
@@ -1612,8 +1612,9 @@ def _initialize_x_y(self, z):
         levels : int or array-like, optional
             Determines the number and positions of the contour lines / regions.
 
-            If an int *n*, use *n* data intervals; i.e. draw *n+1* contour
-            lines. The level heights are automatically chosen.
+            If an int *n*, use at most *n+2* data intervals;
```

Comment:
This sort of thing drives people nuts, though.  I have no idea what "at most" means here.  Is it "usually", but occasionally one gets dropped, or is it usually *n* contours, but occasionally another one will be added.  I know its not your fault, but can we give more information?  I think its also really unfortunate levels=5 will "at most" draw 6 contours.

Reply:
>  I have no idea what "at most" means here.

Since `MaxNLocator` tries to divide the data range into bins of "nice steps" we have to live with the drawback of not being able to control the exact number of bins (and all we can guarantee is the maximum number of bins -- hence I used the words "at most").

https://github.com/matplotlib/matplotlib/blob/68bdd3e0cdefe1db56aeb2e767a64afd98e33b07/lib/matplotlib/ticker.py#L2013

--------------------------------------------------------------------------------
>  I think its also really unfortunate levels=5 will "at most" draw 6 contours.


This issue comes from:
https://github.com/matplotlib/matplotlib/blob/68bdd3e0cdefe1db56aeb2e767a64afd98e33b07/lib/matplotlib/contour.py#L1116

Reply:
based on the suggestion in the original issue, would the following sound better?
`If an int n, use m data intervals, where m <= n+2; i.e. draw m-1 contour lines.`

Reply:
Maybe: "If an int *n*, use an algorithm (see `.MaxNLocator`) that tries to provide no more than n+1 "nice" contour levels between vmin and vmax."  

Including the phrase "nice" will make it clear the algorithm needs some slop....

Reply:
Thanks! that does sound clearer.

Reply:
I'd go for @jklymak's proposal.